In [1]:
import pandas as pd
import numpy as np

total = 2985217

## Load Data

In [2]:
train_2017 = pd.read_csv("train_2017.csv", parse_dates=["transactiondate"])
print(train_2017.shape)
train_2017.tail()

(77613, 3)


parcelid  logerror transactiondate
77608  10833991 -0.002245      2017-09-20
77609  11000655  0.020615      2017-09-20
77610  17239384  0.013209      2017-09-21
77611  12773139  0.037129      2017-09-21
77612  12826780  0.007204      2017-09-25

In [3]:
property_2017 = pd.read_csv("properties_2017.csv")
print(property_2017.shape)
property_2017.tail()

/usr/local/lib/python3.7/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (49) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


(2985217, 58)


parcelid  airconditioningtypeid  architecturalstyletypeid  \
2985212  12170430                    NaN                       NaN   
2985213  11422430                    NaN                       NaN   
2985214  17203030                    NaN                       NaN   
2985215  14242830                    NaN                       NaN   
2985216  12766630                    NaN                       NaN   

         basementsqft  bathroomcnt  bedroomcnt  buildingclasstypeid  \
2985212           NaN          NaN         NaN                  NaN   
2985213           NaN          NaN         NaN                  NaN   
2985214           NaN          NaN         NaN                  NaN   
2985215           NaN          NaN         NaN                  NaN   
2985216           NaN          NaN         NaN                  NaN   

         buildingqualitytypeid  calculatedbathnbr  decktypeid  \
2985212                    NaN                NaN         NaN   
2985213                    NaN                NaN         NaN   
2985214                    NaN                NaN         NaN   
2985215                    NaN                NaN         NaN   
2985216                    NaN                NaN         NaN   

                ...           numberofstories  fireplaceflag  \
2985212         ...                       NaN            NaN   
2985213         ...                       NaN            NaN   
2985214         ...                       NaN            NaN   
2985215         ...                       NaN            NaN   
2985216         ...                       NaN            NaN   

         structuretaxvaluedollarcnt  taxvaluedollarcnt  assessmentyear  \
2985212                         NaN                NaN             NaN   
2985213                         NaN                NaN             NaN   
2985214                         NaN                NaN             NaN   
2985215                         NaN                NaN             NaN   
2985216                         NaN                NaN             NaN   

         landtaxvaluedollarcnt  taxamount  taxdelinquencyflag  \
2985212                    NaN        NaN                 NaN   
2985213                    NaN        NaN                 NaN   
2985214                    NaN        NaN                 NaN   
2985215                    NaN        NaN                 NaN   
2985216                    NaN        NaN                 NaN   

         taxdelinquencyyear  censustractandblock  
2985212                 NaN                  NaN  
2985213                 NaN                  NaN  
2985214                 NaN                  NaN  
2985215                 NaN                  NaN  
2985216                 NaN                  NaN  

[5 rows x 58 columns]

## Dictionary

### airconditioningtypeid
- Type of cooling system present in the home (if any)
- category / 7
- rate of NaN : 2173698 / total ( 72.6% ) 

In [4]:
print(len(property_2017["airconditioningtypeid"].value_counts()))
property_2017["airconditioningtypeid"].value_counts().head()

7


1.0     744291
13.0     58487
5.0       8795
9.0       1905
11.0      1818
Name: airconditioningtypeid, dtype: int64

In [5]:
property_2017["airconditioningtypeid"].isna().sum()/ total

0.7268667570900206

### architecturalstyletypeid
- Architectural style of the home (i.e. ranch, colonial, split-level, etc…)
- category / 8-type
- NaN : 2979156 / 2985217 ( 99.79% )

In [6]:
print(len(property_2017["architecturalstyletypeid"].value_counts()))
property_2017["architecturalstyletypeid"].value_counts().head()

8


7.0     5251
8.0      380
2.0      201
21.0     150
3.0       58
Name: architecturalstyletypeid, dtype: int64

In [7]:
property_2017["architecturalstyletypeid"].isna().sum() / total

0.9979696618369787

### basementsqft
- Finished living area below or partially below ground level
- (scalar/category) / 751
- NaN : 2983589 / 2985217 ( 99.94% )

In [8]:
print(len(property_2017["basementsqft"].value_counts()))
property_2017["basementsqft"].value_counts().head()

751


1528.0    25
700.0     21
240.0     21
100.0     18
144.0     16
Name: basementsqft, dtype: int64

In [9]:
property_2017["basementsqft"].isna().sum() / total

0.9994549809946814

### bathroomcnt
- Number of bathrooms in home including fractional bathrooms
- scalar / 39
- NaN : 11462 / 2985217 ( 0.09% )

In [10]:
print(len(property_2017["bathroomcnt"].value_counts()))
property_2017["bathroomcnt"].value_counts().head()

39


2.0    1219811
3.0     633089
1.0     499332
2.5     208809
4.0     133922
Name: bathroomcnt, dtype: int64

In [11]:
property_2017["bathroomcnt"].isna().sum() / total

0.0009905477558247858

### bedroomcnt
- Number of bedrooms in home
- scalar / 25 
- NaN  : 11450 / total ( 0.09% )

In [12]:
print(len(property_2017["bedroomcnt"].value_counts()))
property_2017["bedroomcnt"].value_counts().head()

25


3.0    1172757
4.0     731475
2.0     606782
5.0     182765
0.0     118705
Name: bedroomcnt, dtype: int64

In [13]:
property_2017["bedroomcnt"].isna().sum() / total

0.0009865279475495417

### buildingclasstypeid
- The building framing type (steel frame, wood frame, concrete/brick) 
- category / 5-types
- NaN : 2972588 / total ( 99.57% ) 

In [14]:
print(len(property_2017["buildingclasstypeid"].value_counts()))
property_2017["buildingclasstypeid"].value_counts().head()

5


4.0    9500
3.0    3028
2.0      84
1.0      75
5.0      44
Name: buildingclasstypeid, dtype: int64

In [15]:
property_2017["buildingclasstypeid"].isna().sum() / total

0.9957353184039888

### buildingqualitytypeid
- Overall assessment of condition of the building from best (lowest) to worst (highest)
- category / 12-types
- NaN : 1056729 / total ( 35% )

In [16]:
print(len(property_2017["buildingqualitytypeid"].value_counts()))
property_2017["buildingqualitytypeid"].value_counts().head()

12


6.0    561502
8.0    501925
4.0    448049
7.0    190244
5.0    107053
Name: buildingqualitytypeid, dtype: int64

In [17]:
property_2017["buildingqualitytypeid"].isna().sum() / total

0.3496636927901724

### calculatedbathnbr
- Number of bathrooms in home including fractional bathroom
- scalar / 36
- NaN : 128912 / total ( 3.9% )

In [18]:
print(len(property_2017["calculatedbathnbr"].value_counts()))
property_2017["calculatedbathnbr"].value_counts().head()

36


2.0    1219799
3.0     633088
1.0     499324
2.5     208578
4.0     133922
Name: calculatedbathnbr, dtype: int64

In [19]:
property_2017["calculatedbathnbr"].isna().sum() / total

0.03924538819121022

### decktypeid
- Type of deck (if any) present on parcel
- category / 1-type
- NaN : 2968121 / total ( 99.42% )

In [20]:
print(len(property_2017["decktypeid"].value_counts()))
property_2017["decktypeid"].value_counts().head()

1


66.0    17379
Name: decktypeid, dtype: int64

In [21]:
property_2017["decktypeid"].isna().sum() / total

0.9941783126653774

### threequarterbathnbr
- Number of 3/4 bathrooms in house (shower + sink + toilet)
- scalar / 7개
- NaN : 2673586 / total ( 89% )

In [22]:
print(len(property_2017["threequarterbathnbr"].value_counts()))
property_2017["threequarterbathnbr"].value_counts().head()

7


1.0    313464
2.0      2548
3.0       268
4.0        48
5.0        18
Name: threequarterbathnbr, dtype: int64

In [23]:
property_2017["threequarterbathnbr"].isna().sum() / total

0.8940254594557112

### finishedfloor1squarefeet
- Size of the finished living area on the first (entry) floor of the home
- scalar / 4625 
- NaN : 2782500 / total ( 93.20% )

In [24]:
print(len(property_2017["finishedfloor1squarefeet"].value_counts()))
property_2017["finishedfloor1squarefeet"].value_counts().head()

4625


1252.0    923
1233.0    892
1120.0    705
1226.0    616
1612.0    597
Name: finishedfloor1squarefeet, dtype: int64

In [25]:
property_2017["finishedfloor1squarefeet"].isna().sum() / total

0.9317443254543974

### calculatedfinishedsquarefeet
- Calculated total finished living area of the home
- scalar / 12631
- NaN : 55565 (1.5%)

In [26]:
print(len(property_2017["calculatedfinishedsquarefeet"].value_counts()))
property_2017["calculatedfinishedsquarefeet"].value_counts().head()

12631


1440.0    8968
1344.0    7211
1200.0    7111
960.0     6024
1120.0    5956
Name: calculatedfinishedsquarefeet, dtype: int64

In [27]:
property_2017["calculatedfinishedsquarefeet"].isna().sum() / total

0.015106774482391063

### finishedsquarefeet6
- Base unfinished and finished area
- scalar / 4221
- NaN : 2963216 / total ( 99.28% )

In [28]:
print(len(property_2017["finishedsquarefeet6"].value_counts()))
property_2017["finishedsquarefeet6"].value_counts().head()

4221


952.0     101
800.0      90
3820.0     88
720.0      74
600.0      66
Name: finishedsquarefeet6, dtype: int64

In [29]:
property_2017["finishedsquarefeet6"].isna().sum() / total

0.9928038732192668

### finishedsquarefeet12
- Finished living area
- scalar / 10732
- NaN : 276033 / total ( 8.8% )

In [30]:
print(len(property_2017["finishedsquarefeet12"].value_counts()))
property_2017["finishedsquarefeet12"].value_counts().head()

10732


1440.0    6965
1200.0    6645
1344.0    6243
1120.0    5698
1080.0    5625
Name: finishedsquarefeet12, dtype: int64

In [31]:
property_2017["finishedsquarefeet12"].isna().sum() / total

0.08858016016926072

### finishedsquarefeet13
- Perimeter  living area
- scalar / 250
- NaN : 2977545 / total ( 99.74% )

In [32]:
print(len(property_2017["finishedsquarefeet13"].value_counts()))
property_2017["finishedsquarefeet13"].value_counts().head()

250


1440.0    1571
1344.0     709
1248.0     453
1536.0     355
720.0      245
Name: finishedsquarefeet13, dtype: int64

In [33]:
property_2017["finishedsquarefeet13"].isna().sum() / total

0.9974303375600501

### finishedsquarefeet15
- Total area
- scalar / 8798
- NaN : 2794419 / total ( 93.6% )

In [34]:
print(len(property_2017["finishedsquarefeet15"].value_counts()))
property_2017["finishedsquarefeet15"].value_counts().head()

8798


1536.0    649
1728.0    430
1600.0    415
1440.0    412
1800.0    401
Name: finishedsquarefeet15, dtype: int64

In [35]:
property_2017["finishedsquarefeet15"].isna().sum() / total

0.9362910635977217

### finishedsquarefeet50
-  Size of the finished living area on the first (entry) floor of the home
- scalar / 4888
- rate of NaN : 93.2 %

In [36]:
print(len(property_2017["finishedsquarefeet50"].value_counts()))
property_2017["finishedsquarefeet50"].value_counts().head()

4888


1252.0    923
1233.0    891
1120.0    705
1226.0    616
1612.0    596
Name: finishedsquarefeet50, dtype: int64

In [37]:
property_2017["finishedsquarefeet50"].isna().sum() / total

0.9317443254543974

### fips
-  Federal Information Processing Standard code -  see https://en.wikipedia.org/wiki/FIPS_county_code for more details
- category / 3
- rate of NaN : 0.09%

In [38]:
print(len(property_2017["fips"].value_counts()))
property_2017["fips"].value_counts().head()

3


6037.0    2012741
6059.0     745800
6111.0     223744
Name: fips, dtype: int64

In [39]:
property_2017["fips"].isna().sum() / total

0.0009821731552513602

### fireplacecnt
-  Number of fireplaces in a home (if any)
- scalar / 9
- rate of NaN : 89.5%

In [40]:
print(len(property_2017["fireplacecnt"].value_counts()))
property_2017["fireplacecnt"].value_counts().head()

9


1.0    270019
2.0     34487
3.0      7716
4.0       716
5.0       129
Name: fireplacecnt, dtype: int64

In [41]:
property_2017["fireplacecnt"].isna().sum() / total

0.8951084628018666

### fireplaceflag
- Is a fireplace present in this home 
- category / 2 (True / False)
- rate of NaN : 99.8%

In [42]:
print(len(property_2017["fireplaceflag"].value_counts()))
property_2017["fireplaceflag"].value_counts().head()

1


True    5163
Name: fireplaceflag, dtype: int64

In [43]:
property_2017["fireplaceflag"].isna().sum() / total

0.9982704774895761

### fullbathcnt
- Number of full bathrooms (sink, shower + bathtub, and toilet) present in home
- scalar / 22
- rate of NaN : 3.9%

In [44]:
print(len(property_2017["fullbathcnt"].value_counts()))
property_2017["fullbathcnt"].value_counts().head()

22


2.0    1428927
3.0     664914
1.0     544794
4.0     153684
5.0      44721
Name: fullbathcnt, dtype: int64

In [45]:
property_2017["fullbathcnt"].isna().sum() / total

0.03924538819121022

### garagecarcnt
- Total number of garages on the lot including an attached garage
- scalar / 24
- rate of NaN : 70%

In [46]:
print(len(property_2017["garagecarcnt"].value_counts()))
property_2017["garagecarcnt"].value_counts().head()

24


2.0    667830
1.0    177901
3.0     19884
0.0     13583
4.0      8655
Name: garagecarcnt, dtype: int64

In [47]:
property_2017["garagecarcnt"].isna().sum() / total

0.7015265556909264

### garagetotalsqft
- Total number of square feet of all garages on lot including an attached garage
- scalar / 2253
- rate of NaN : 70%

In [48]:
print(len(property_2017["garagetotalsqft"].value_counts()))
property_2017["garagetotalsqft"].value_counts().head()

2253


0.0      200050
400.0     33497
360.0     29706
441.0     26834
462.0     25514
Name: garagetotalsqft, dtype: int64

In [49]:
property_2017["garagetotalsqft"].isna().sum() / total

0.7015265556909264

### hashottuborspa
- Does the home have a hot tub or spa
- category / 2 (True/False)
- rate of NaN : 98.3%

In [50]:
print(len(property_2017["hashottuborspa"].value_counts()))
property_2017["hashottuborspa"].value_counts().head()

1


True    50062
Name: hashottuborspa, dtype: int64

In [51]:
property_2017["hashottuborspa"].isna().sum() / total

0.9832300298437266

### heatingorsystemtypeid
- Type of home heating system
- category / 14
- rate of NaN : 37.4%

In [52]:
print(len(property_2017["heatingorsystemtypeid"].value_counts()))
property_2017["heatingorsystemtypeid"].value_counts().head()

14


2.0     1168610
7.0      646021
6.0       27481
24.0      21107
20.0       3630
Name: heatingorsystemtypeid, dtype: int64

In [53]:
property_2017["heatingorsystemtypeid"].isna().sum() / total

0.3738599237509367

### latitude
- Latitude of the middle of the parcel multiplied by 10e6
- scalar / 852718
- rate of NaN : 0.09%

In [54]:
print(len(property_2017["latitude"].value_counts()))
property_2017["latitude"].value_counts().head()

852718


33984600.0    859
33708500.0    756
34056700.0    740
34171800.0    611
33766300.0    568
Name: latitude, dtype: int64

In [55]:
property_2017["latitude"].isna().sum() / total

0.0009821731552513602

### longitude
-  Longitude of the middle of the parcel multiplied by 10e6
- scalar / 1043238
- rate of NaN : 0.09%

In [56]:
print(len(property_2017["longitude"].value_counts()))
property_2017["longitude"].value_counts().head()

1043238


-118390000.0    1718
-118438000.0    1585
-118389000.0    1569
-118433000.0    1545
-118252000.0    1402
Name: longitude, dtype: int64

In [57]:
property_2017["longitude"].isna().sum() / total

0.0009821731552513602

### lotsizesquarefeet
- Area of the lot in square feet
- scalar / 70214
- rate of NaN : 9.1%

In [58]:
print(len(property_2017["lotsizesquarefeet"].value_counts()))
property_2017["lotsizesquarefeet"].value_counts().head()

70214


6000.0    44585
5000.0    17517
7200.0    15326
7000.0    11864
6500.0    11565
Name: lotsizesquarefeet, dtype: int64

In [59]:
property_2017["lotsizesquarefeet"].isna().sum() / total

0.09135215295906461

### numberofstories
-  Number of stories or levels the home has
- scalar / 12
- rate of NaN : 77%

In [60]:
print(len(property_2017["numberofstories"].value_counts()))
property_2017["numberofstories"].value_counts().head()

12


1.0    426779
2.0    243078
3.0     15721
4.0        69
5.0        18
Name: numberofstories, dtype: int64

In [61]:
property_2017["numberofstories"].isna().sum() / total

0.7703094950886318

### poolcnt
- Number of pools on the lot (if any)
- scalar / 1
- rate of NaN : 82%

In [62]:
print(len(property_2017["poolcnt"].value_counts()))
property_2017["poolcnt"].value_counts().head()

1


1.0    539632
Name: poolcnt, dtype: int64

In [63]:
property_2017["poolcnt"].isna().sum() / total

0.8192319017344468

### poolsizesum
- Total square footage of all pools on property
- scalar / 907
- rate of NaN : 99%

In [64]:
print(len(property_2017["poolsizesum"].value_counts()))
property_2017["poolsizesum"].value_counts().head()

907


450.0    3179
400.0     972
480.0     917
500.0     828
512.0     826
Name: poolsizesum, dtype: int64

In [65]:
property_2017["poolsizesum"].isna().sum() / total

0.9906345166867266

### pooltypeid10
- Spa or Hot Tub
- category / 1
- rate of NaN : 99.4%

In [66]:
print(len(property_2017["pooltypeid10"].value_counts()))
property_2017["pooltypeid10"].value_counts().head()

1


1.0    17006
Name: pooltypeid10, dtype: int64

In [67]:
property_2017["pooltypeid10"].isna().sum() / total

0.994303261705933

### pooltypeid2
- Pool with Spa/Hot Tub
- cateory / 1
- rate of NaN : 98.9%

In [68]:
print(len(property_2017["pooltypeid2"].value_counts()))
property_2017["pooltypeid2"].value_counts().head()

1


1.0    33056
Name: pooltypeid2, dtype: int64

In [69]:
property_2017["pooltypeid2"].isna().sum() / total

0.9889267681377937

### pooltypeid7
- Pool without hot tub
- category / 1
- rate of NaN : 83%

In [70]:
print(len(property_2017["pooltypeid7"].value_counts()))
property_2017["pooltypeid7"].value_counts().head()

1


1.0    505895
Name: pooltypeid7, dtype: int64

In [71]:
property_2017["pooltypeid7"].isna().sum() / total

0.8305332577162732

### propertycountylandusecode
- County land use code i.e. it's zoning at the county level
- category(object) / 234
- rate of NaN : 0.1%

In [72]:
print(len(property_2017["propertycountylandusecode"].value_counts()))
property_2017["propertycountylandusecode"].value_counts().head()

234


0100    1153332
122      522127
0101     248386
010C     225950
1111     127012
Name: propertycountylandusecode, dtype: int64

In [73]:
property_2017["propertycountylandusecode"].isna().sum() / total

0.001004617084788141

### propertylandusetypeid
- Type of land use the property is zoned for
- category / 16
- rate of NaN : 0.09%

In [74]:
print(len(property_2017["propertylandusetypeid"].value_counts()))
property_2017["propertylandusetypeid"].value_counts().head()

16


261.0    2152863
266.0     483789
246.0     114415
269.0      61559
263.0      59344
Name: propertylandusetypeid, dtype: int64

In [75]:
property_2017["propertylandusetypeid"].isna().sum() / total

0.0009821731552513602

### propertyzoningdesc
- Description of the allowed land uses (zoning) for that property
- category(object) / 5651
- rate of NaN : 33.5%

In [76]:
print(len(property_2017["propertyzoningdesc"].value_counts()))
property_2017["propertyzoningdesc"].value_counts().head()

5651


LAR1     275246
LAR3      67089
LARS      54862
LBR1N     52750
LAR2      48858
Name: propertyzoningdesc, dtype: int64

In [77]:
property_2017["propertyzoningdesc"].isna().sum() / total

0.3359038890640111

### rawcensustractandblock
- Census tract and block ID combined - also contains blockgroup assignment by extension
- category / 99513
- rate of NaN : 0.09%

In [78]:
print(len(property_2017["rawcensustractandblock"].value_counts()))
property_2017["rawcensustractandblock"].value_counts().head()

99513


6.059052e+07    1576
6.037920e+07    1455
6.059052e+07     971
6.059063e+07     895
6.037137e+07     886
Name: rawcensustractandblock, dtype: int64

In [79]:
property_2017["rawcensustractandblock"].isna().sum() / total

0.0009821731552513602

### censustractandblock
- Census tract and block ID combined - also contains blockgroup assignment by extension
- category / 96755
- rate of NaN : 2.5%

In [80]:
print(len(property_2017["censustractandblock"].value_counts()))
property_2017["censustractandblock"].value_counts().head()

96755


6.037920e+13    1171
6.037137e+13     881
6.037920e+13     811
6.059063e+13     807
6.059052e+13     806
Name: censustractandblock, dtype: int64

In [81]:
property_2017["censustractandblock"].isna().sum() / total

0.025118776959932896

### regionidcounty
- County in which the property is located
- category / 3
- rate of NaN : 0.09%

In [82]:
print(len(property_2017["regionidcounty"].value_counts()))
property_2017["regionidcounty"].value_counts().head()

3


3101.0    2012741
1286.0     745800
2061.0     223744
Name: regionidcounty, dtype: int64

In [83]:
property_2017["regionidcounty"].isna().sum() / total

0.0009821731552513602

### regionidcity
- City in which the property is located (if any)
- category / 186
- rate of NaN : 2%

In [84]:
print(len(property_2017["regionidcity"].value_counts()))
property_2017["regionidcity"].value_counts().head()

186


12447.0    670868
46298.0     93454
16764.0     59793
52650.0     58619
25218.0     57013
Name: regionidcity, dtype: int64

In [85]:
property_2017["regionidcity"].isna().sum() / total

0.020811887377031553

### regionidzip
- Zip code in which the property is located
- category / 403
- rate of NaN : 0.4%

In [86]:
print(len(property_2017["regionidzip"].value_counts()))
property_2017["regionidzip"].value_counts().head()

403


96987.0    22025
96193.0    21756
97118.0    20615
97319.0    20337
96964.0    19923
Name: regionidzip, dtype: int64

In [87]:
property_2017["regionidzip"].isna().sum() / total

0.004258986867621349

### regionidneighborhood
- Neighborhood in which the property is located
- category / 529
- rate of NaN : 61%

In [88]:
print(len(property_2017["regionidneighborhood"].value_counts()))
property_2017["regionidneighborhood"].value_counts().head()

529


118208.0    32244
268496.0    23182
48570.0     21188
27080.0     18883
37739.0     18648
Name: regionidneighborhood, dtype: int64

In [89]:
property_2017["regionidneighborhood"].isna().sum() / total

0.6125102463238016

### roomcnt
- Total number of rooms in the principal residence
- scalar / 36
- rate of NaN : 0.09%

In [90]:
print(len(property_2017["roomcnt"].value_counts()))
property_2017["roomcnt"].value_counts().head()

36


0.0    2315644
6.0     178563
7.0     156659
8.0     120448
5.0      99734
Name: roomcnt, dtype: int64

In [91]:
property_2017["roomcnt"].isna().sum() / total

0.0009945675641000302

### storytypeid
- Type of floors in a multi-story house (i.e. basement and main level, split-level, attic, etc.).  See tab for details.
- category / 1
- rate of NaN : 99.9%

In [92]:
print(len(property_2017["storytypeid"].value_counts()))
property_2017["storytypeid"].value_counts().head()

1


7.0    1623
Name: storytypeid, dtype: int64

In [93]:
property_2017["storytypeid"].isna().sum() / total

0.9994563209307732

### typeconstructiontypeid
- What type of construction material was used to construct the home
- category / 5
- rate of NaN : 99.7%

In [94]:
print(len(property_2017["typeconstructiontypeid"].value_counts()))
property_2017["typeconstructiontypeid"].value_counts().head()

5


6.0     6669
4.0       59
13.0      14
10.0       3
11.0       1
Name: typeconstructiontypeid, dtype: int64

In [95]:
property_2017["typeconstructiontypeid"].isna().sum() / total

0.9977401977812668

### unitcnt
-  Number of units the structure is built into (i.e. 2 = duplex, 3 = triplex, etc...)
- category / 154
- rate of NaN : 33.6%

In [96]:
print(len(property_2017["unitcnt"].value_counts()))
property_2017["unitcnt"].value_counts().head()

154


1.0    1783873
2.0     115778
4.0      39926
3.0      39893
5.0        325
Name: unitcnt, dtype: int64

In [97]:
property_2017["unitcnt"].isna().sum() / total

0.3363825812327881

### yardbuildingsqft17
- Patio in  yard
- scalar / 1664
- rate of NaN : 97.2%

In [98]:
print(len(property_2017["yardbuildingsqft17"].value_counts()))
property_2017["yardbuildingsqft17"].value_counts().head()

1664


240.0    3202
200.0    3083
300.0    2278
180.0    1509
288.0    1452
Name: yardbuildingsqft17, dtype: int64

In [99]:
property_2017["yardbuildingsqft17"].isna().sum() / total

0.972669323536614

### yardbuildingsqft26
- Storage shed/building in yard
- scalar / 594
- rate of NaN : 99.9%

In [100]:
print(len(property_2017["yardbuildingsqft26"].value_counts()))
property_2017["yardbuildingsqft26"].value_counts().head()

594


120.0    91
100.0    69
240.0    61
160.0    55
80.0     53
Name: yardbuildingsqft26, dtype: int64

In [101]:
property_2017["yardbuildingsqft26"].isna().sum() / total

0.9991136322753086

### yearbuilt
- The Year the principal residence was built 
- category(also use scalar) / 183
- rate of NaN :  1.6%

In [102]:
print(len(property_2017["yearbuilt"].value_counts()))
property_2017["yearbuilt"].value_counts().head()

183


1955.0    88555
1950.0    76100
1954.0    73284
1953.0    67062
1956.0    65723
Name: yearbuilt, dtype: int64

In [103]:
property_2017["yearbuilt"].isna().sum() / total

0.016023290769146766

### taxvaluedollarcnt
- The total tax assessed value of the parcel
- scalar / 661520
- rate of NaN : 1.1% 

In [104]:
print(len(property_2017["taxvaluedollarcnt"].value_counts()))
property_2017["taxvaluedollarcnt"].value_counts().head()

661520


9.0         1902
450000.0    1452
400000.0    1371
500000.0    1285
480000.0    1131
Name: taxvaluedollarcnt, dtype: int64

In [105]:
property_2017["taxvaluedollarcnt"].isna().sum() / total

0.011478562529960134

### structuretaxvaluedollarcnt
- The assessed value of the built structure on the parcel
- scalar / 437542
- rate of NaN : 1.5%

In [106]:
print(len(property_2017["structuretaxvaluedollarcnt"].value_counts()))
property_2017["structuretaxvaluedollarcnt"].value_counts().head()

437542


100000.0    1562
104023.0    1385
106103.0    1371
109038.0    1324
108225.0    1197
Name: structuretaxvaluedollarcnt, dtype: int64

In [107]:
property_2017["structuretaxvaluedollarcnt"].isna().sum() / total

0.015564697641745976

### landtaxvaluedollarcnt
- The assessed value of the land area of the parcel
- scalar / 553303
- rate of NaN : 2%

In [108]:
print(len(property_2017["landtaxvaluedollarcnt"].value_counts()))
property_2017["landtaxvaluedollarcnt"].value_counts().head()

553303


21299.0    3643
22755.0    2369
16522.0    2298
9.0        2184
19441.0    2127
Name: landtaxvaluedollarcnt, dtype: int64

In [109]:
property_2017["landtaxvaluedollarcnt"].isna().sum() / total

0.02007425255852422

### taxamount
- The total property tax assessed for that assessment year
- scalar / 1050011
- rate of NaN : 0.7%

In [110]:
print(len(property_2017["taxamount"].value_counts()))
property_2017["taxamount"].value_counts().head()

1050011


345.72     173
1844.16    124
1045.88    111
1481.24    101
2264.43     89
Name: taxamount, dtype: int64

In [111]:
property_2017["taxamount"].isna().sum() / total

0.007621556489863216

### assessmentyear
- The year of the property tax assessment 
- category / 14
- rate of NaN : 0.09%

In [112]:
print(len(property_2017["assessmentyear"].value_counts()))
property_2017["assessmentyear"].value_counts().head()

14


2016.0    2978601
2015.0       2342
2014.0       1290
2012.0         26
2004.0          6
Name: assessmentyear, dtype: int64

In [113]:
property_2017["assessmentyear"].isna().sum() / total

0.0009825081392742974

### taxdelinquencyflag
- Property taxes for this parcel are past due as of 2015
- category / 2 (Y/N)
- rate of NaN : 98%

In [114]:
print(len(property_2017["taxdelinquencyflag"].value_counts()))
property_2017["taxdelinquencyflag"].value_counts().head()

1


Y    56515
Name: taxdelinquencyflag, dtype: int64

In [115]:
property_2017["taxdelinquencyflag"].isna().sum() / total

0.9810683779437139

### taxdelinquencyyear
- Year for which the unpaid propert taxes were due 
- category / 31
- rate of NaN : 98%

In [116]:
print(len(property_2017["taxdelinquencyyear"].value_counts()))
property_2017["taxdelinquencyyear"].value_counts().head()

31


15.0    26105
14.0    16838
13.0     4634
12.0     3126
11.0     2350
Name: taxdelinquencyyear, dtype: int64

In [117]:
property_2017["taxdelinquencyyear"].isna().sum() / total

0.9810677079756681